### PR TITLE
Add tests for SQL-backed state impl

### PR DESCRIPTION
**Pull request by mheon**
Minor changes to container.go and sql_state.go to fix issues
identified by the tests

### CODE REVIEW

**Comment by mheon** [comment]
@rhatdan PTAL

File: libpod/sql_state_test.go [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+
+func getTestContainer(id, name string) *Container {
+	ctr := &Container{
+		config: &containerConfig{
+			ID:              id,
+			Name:            name,
+			RootfsImageID:   id,
+			RootfsImageName: "testimg",
+			UseImageConfig:  true,
+			StaticDir:       "/var/lib/libpod/",
```
Comment (rhatdan) [inline libpod/sql_state_test.go]: Lets keep this in containers /var/lib/containers/libpod?

Comment (rhatdan) [reply libpod/sql_state_test.go]: Actually probably should use tmpdirs for these.

**Comment by rhatdan** [comment]
LGTM
Fix travis issues and don't use data on disk.

File: libpod/sql_state_test.go (hunk 1 above)
Comment (mheon) [reply libpod/sql_state_test.go]: This isn't actually used, it's just temporary paths to check that the container is being retrieved correctly. Let me make that clear

**Comment by rhatdan** [comment]
Merging.

**Pull request merged**